src/main/java/org/wildfly/security/auth/provider/SimpleMapBackedSecurityRealm.java

```diff
@@ -118,5 +121,17 @@ public CredentialSupport getCredentialSupport(Class<?> credentialType) {
             final Password password = map.get(principal);
             return credentialType.isInstance(password) ? credentialType.cast(password) : null;
         }
+
+        public VerificationResult verifyCredential(final Object credential) {
```

Comment:
If I leave out he ones that could be invalid this is the list I end up with: -

UNSUPPORTED  (Don't support in any way)
POSSIBLY_VERIFIABLE (Can't obtain but might be able to verify)
VERIFIABLE (Not obtainable)
UNKNOWN (Might be able to obtain or verify, nothing known for certain)
VERIFIABLE_MAYBE_OBTAINABLE (Definitely verifiable, maybe obtainable)
SUPPORTED (Obtainable and verifiable)

This excludes: -
  Possibly obtainable, not verifiable.
  Obtainable but not verifiable
  Obtainable but possibly verifiable.

I think the last one is invalid in all situations, the first two are invalid if we mandate that realms do implement verification if they can.